### PR TITLE
feat(ephemeralcontainer): use ephemeral container to kill sidecar.

### PR DESCRIPTION
**Pull request by justbert**
Implementation to allow for the termination of sidecars using ephemeral containers.

### CODE REVIEW

File: pkg/sidecarterminator/helpers.go [hunk 1]
```diff
@@ -64,3 +64,17 @@ func isCompleted(pod *v1.Pod, sidecars map[string]int) bool {
 
 	return false
 }
+
+func hasSidecarTerminator(pod *v1.Pod, sidecar v1.ContainerStatus) bool {
```
Comment (zachomedia) [inline pkg/sidecarterminator/helpers.go]:
```suggestion
func hasSidecarTerminatorContainer(pod *v1.Pod, sidecar v1.ContainerStatus) bool {
```

File: pkg/sidecarterminator/helpers.go [hunk 2]
```diff
@@ -64,3 +64,17 @@ func isCompleted(pod *v1.Pod, sidecars map[string]int) bool {
...
+	for _, ephCont := range pod.Spec.EphemeralContainers {
+		if ephCont.Name == generateSidecarTerminatorName(sidecar.Name) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func generateSidecarTerminatorName(sidecarName string) string {
```
Comment (zachomedia) [inline pkg/sidecarterminator/helpers.go]:
```suggestion
func generateSidecarTerminatorContainerName(sidecarName string) string {
```

File: pkg/sidecarterminator/terminator.go [hunk 3]
```diff
@@ -4,35 +4,37 @@ import (
...
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/remotecommand"
 	"k8s.io/klog"
 )
 
+const (
+	SIDECAR_TERMINATOR_CONTAINER = "sidecar-terminator"
```
Comment (zachomedia) [inline pkg/sidecarterminator/terminator.go]:
```suggestion
	SidecarTerminatorContainerNamePrefix = "sidecar-terminator"
```

The Go standard is for camel case names on consts. Also might be good to clarify it's part of the name prefix.

**Review by zachomedia — CHANGES_REQUESTED** [review]
A few stylistic points, but otherwise this is looking great!

File: pkg/sidecarterminator/helpers.go (hunk 1 above)
Comment (justbert) [reply pkg/sidecarterminator/helpers.go]: Done!

File: pkg/sidecarterminator/terminator.go (hunk 3 above)
Comment (justbert) [reply pkg/sidecarterminator/terminator.go]: Thanks! Duh on my part!

File: pkg/sidecarterminator/helpers.go (hunk 2 above)
Comment (justbert) [reply pkg/sidecarterminator/helpers.go]: Done!

**Review by zachomedia — APPROVED** [review]
Thanks! This looks ready to go :)

**Pull request merged**